### ISSUE TITLE
Allow setting proxy for the EPEL module

### ISSUE THREAD

**Issue by adamjk-dev** [justindowning/puppet-awscli#38]
Right now, the module does an 'include epel', but for those who have to set a proxy, they need to be able to pass that param into that module.

What I had to do:
1. deps.pp:

```
class awscli::deps (
  $proxy = $awscli::params::proxy,
) inherits awscli::params {
  case $::osfamily {
    'Debian': {
      include awscli::deps::debian
    }
    'RedHat': {
      class { 'awscli::deps::redhat':
        proxy => $proxy,
      }
    }
    'Darwin': {
      include awscli::deps::osx
    }
    default:  { fail("The awscli module does not support ${::osfamily}") }
  }
}
```
1. redhat.pp:

```
class awscli::deps::redhat (
  $proxy = $awscli::params::proxy,
) inherits awscli::params {
  # Check if we have a proxy to setup with EPEL
  if $proxy != undef {
    class { '::epel':
      epel_proxy => $proxy,
    }
  }
  else {
    include ::epel
  }
  Package { require => Class['epel'] }

  if $awscli::install_pkgdeps {
    if ! defined(Package[ $awscli::pkg_dev ]) {
      package { $awscli::pkg_dev: ensure => installed }
    }
  }
  if $awscli::install_pip {
    if ! defined(Package[ $awscli::pkg_pip ]) {
      package { $awscli::pkg_pip: ensure => installed }
    }
  }
}
```
1. init.pp:

```
class awscli (
  $version         = 'present',
  $pkg_dev         = $awscli::params::pkg_dev,
  $pkg_pip         = $awscli::params::pkg_pip,
  $install_pkgdeps = true,
  $install_pip     = true,
  $proxy           = $awscli::params::proxy,
) inherits awscli::params {
  # Handle dependencies
  class { 'awscli::deps':
    proxy => $proxy,
  }
```
1. params.pp:

```
class awscli::params {

  $proxy = undef

  case $::osfamily {
```

**Comment by justindowning** [comment]
Looks good. Want to PR this?

**Comment by justindowning** [comment]
Fixed in #39
